src/lib/HoneylemonService.js

```diff
@@ -370,19 +366,168 @@ class HoneylemonService {
       .call();
   }
 
+  async deployDSProxyContract(deployer) {
+    const address = await this.marketContractProxy.methods
+      .createDSProxyWallet()
+      .call({ from: deployer });
+    await this.marketContractProxy.methods.createDSProxyWallet().send({
```

Comment:
looks good, but we can use also `addressToDSProxy(deployer)` to get the user's proxy wallet.

Reply:
so rather deploy first then call `addressToDSProxy(deployer)`? you could do this but then you need to wait for the chain to update and the block the be mined before the function will return. This method will rather return before the block is mined.

Reply:
I think it is the same, as u currently waiting for the `.send()` tx, so in both cases, there is a call tx and a send tx and we will wait for one.
Also is there a case where `const address = await this.marketContractProxy.methods....call(...)` will return an address, then the send transaction will create a wallet with a different address for that user ? like in the case where ur first call tx return add `0x1..`, then another tx get sent from the marketContractProxy before the `createDSProxyWallet().send()`, then the `createDSProxyWallet()` happen, so in this case the nonce change, it will return a diff address, right ? as the created contract address get calculated through the sender nonce...